src/configuration.ts

```diff
@@ -120,8 +120,83 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config = JSON.parse(fileData) as IConfigFile;
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames) {
```

Comment:
In general in TSLint code, we use explicit checks for `null`/`undefined` instead of implicit ones. So instead of `if (parentConfigNames)` we use `if (parentConfigNames != null)`. Or instead of `if (!someVar)` we would use `if (someVar == null)`.  This removes the weirdness of empty strings, `false`, etc being handled incorrectly and is more explicit overall.